crates/weaver_semconv/src/any_value.rs

```diff
@@ -0,0 +1,690 @@
+pub struct AnyValueCommonSpec {
+    /// String that uniquely identifies the enum entry.
+    pub id: String,
+    /// A brief description of the value
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub brief: String,
+    /// A more elaborate description of the value.
+    /// It defaults to an empty string.
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub note: String,
+    /// Specifies the stability of the value.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stability: Option<Stability>,
+    /// Sequence of examples for the value or single example
+    /// value. If only a single example is provided, it can
+    /// directly be reported without encapsulating it
+    /// into a sequence/dictionary.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub examples: Option<Examples>,
+    /// Specifies if the field is mandatory. Can be "required",
+    /// "conditionally_required", "recommended" or "opt_in". When omitted,
+    /// the field is "recommended". When set to
+    /// "conditionally_required", the string provided as <condition> MUST
+    /// specify the conditions under which the field is required.
+    pub requirement_level: RequirementLevel,
+    /// Specifies if the body field is deprecated. The string
+    /// provided as <description> MUST specify why it's deprecated and/or what
+    /// to use instead. See also stability.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub deprecated: Option<String>,
+}
+
+/// Implements a human readable display for AnyValueType.
+impl Display for AnyValueSpec {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            AnyValueSpec::Map { fields, .. } => {
+                let entries = fields
+                    .iter()
+                    .map(|v| format!("{}", v))
+                    .collect::<Vec<String>>()
+                    .join(", ");
+                write!(f, "map<{}>{{ {} }}", self.id(), entries)
+            }
+            AnyValueSpec::Boolean { .. } => write!(f, "boolean"),
+            AnyValueSpec::Int { .. } => write!(f, "int"),
+            AnyValueSpec::Double { .. } => write!(f, "double"),
+            AnyValueSpec::String { .. } => write!(f, "string"),
+            AnyValueSpec::Strings { .. } => write!(f, "string[]"),
+            AnyValueSpec::Ints { .. } => write!(f, "int[]"),
+            AnyValueSpec::Doubles { .. } => write!(f, "double[]"),
+            AnyValueSpec::Booleans { .. } => write!(f, "boolean[]"),
+            AnyValueSpec::Bytes { .. } => write!(f, "byte[]"),
+            AnyValueSpec::Undefined { .. } => write!(f, "undefined"),
+            AnyValueSpec::Enum { .. } => write!(f, "enum<{}>", self.id()),
+        }
+    }
+}
+
+impl AnyValueSpec {
+    /// Returns the common value specification for each type.
+    #[must_use]
+    pub fn common(&self) -> &AnyValueCommonSpec {
```

Comment:
This is used to avoid having this selection "everywhere" especially when resolving the Spec (that part is not included in this PR -- still working on tests in the Events PR)